apps/aecore/src/aec_block_candidate.erl

```diff
@@ -183,14 +184,16 @@ int_create_block(PrevBlockHash, PrevBlock, Trees, Miner, Txs) ->
     Height = PrevBlockHeight + 1,
     Version = aec_hard_forks:protocol_effective_at_height(Height),
 
-    {ok, Txs1, Trees2, Fees} =
+    {ok, ValidTxs, InvalidTxs, Trees2, Fees} =
         int_apply_block_txs(Txs, Miner, Trees, Height, Version, false),
 
-    TxsTree = aec_txs_trees:from_txs(Txs1),
+    aec_tx_pool:invalid_txs([ aetx_sign:hash(STx) || STx <- InvalidTxs ], Height),
```

Comment:
Reading this line it feels it does not fit the responsibility of the exported function. Maybe `InvalidTxs` shall be returned by this function, in turn returned by the only caller function used by the system that is `aec_block_candidate:create/1`, and this line of code shall be in `aec_block_generator:create_block_candidate/1`? Totally optional.

----

I am trying to understand synchronous/asynchronous operations for potential intermittent test failures introduced by this PR (still getting my head around it). This is where this comment is coming from.

Reply:
Hmm, yes, good point. Since `aec_tx_pool:invalid_txs` is asynchronous it should not make a differenc functionality-wise but doing it here is definitely slightly lazy approach!